cdap-api/src/main/java/co/cask/cdap/api/flow/FlowletDefinition.java

```diff
@@ -79,9 +74,24 @@ public FlowletDefinition(String flowletName, Flowlet flowlet, int instances) {
                                                        flowletSpec.getDescription(), flowletSpec.getFailurePolicy(),
                                                        flowletSpec.getDataSets(), flowletSpec.getProperties(),
                                                        flowletSpec.getResources());
-    this.streams = Collections.unmodifiableMap(flowletConfigurer.getStreams());
-    this.datasetModules = Collections.unmodifiableMap(flowletConfigurer.getDatasetModules());
-    this.datasetSpecs = Collections.unmodifiableMap(flowletConfigurer.getDatasetSpecs());
+  }
+
+  public FlowletDefinition(String flowletName, Flowlet flowlet,
```

Comment:
do we need both this constructor and the one above? Seems like we should remove the one above and move DefaultFlowletConfigurer out of cdap-api and into cdap-app-fabric.